editor/src/java/com/defold/util/SupportPath.java

```diff
@@ -87,16 +87,14 @@ public static Path getWindowsSupportPath(String applicationName) {
     // linux
 
     private static Path getLinuxSupportPath(String applicationName) {
-        final String userHome = System.getProperty("user.home");
-        if (userHome == null) {
-            return null;
+        final String xdgStateHome = System.getenv("XDG_STATE_HOME");
+        if (xdgStateHome != null) {
+            // 1. $XDG_STATE_HOME
+            return Path.of(xdgStateHome, applicationName);
+        } else {
+            // 2. ~/.local/state
+            return Path.of(System.getProperty("user.home"), ".local", "state", applicationName);
         }
```

Comment:
Nit, but I guess the else is not needed now. Feel free to leave this as it is.

```suggestion
        }

        // 2. ~/.local/state
        return Path.of(System.getProperty("user.home"), ".local", "state", applicationName);
```